src/grt/src/fastroute/src/maze.cpp

```diff
@@ -2254,23 +2259,64 @@ void FastRouteCore::setCongestionNets(int& posX, int& posY, int dir)
         }
         if (gridsX[i] == gridsX[i + 1]) {  // a vertical edge
           const int ymin = std::min(gridsY[i], gridsY[i + 1]);
-          if (ymin == posY && gridsX[i] == posX && dir == 0) {
-            congestion_nets_.insert(nets_[netID]->getDbNet());
+          if (abs(ymin - posY) <= ratio && abs(gridsX[i] - posX) <= ratio
+              && dir == 0) {
+            congestion_nets.insert(nets_[netID]->getDbNet());
           }
         } else if (gridsY[i] == gridsY[i + 1]) {  // a horizontal edge
           const int xmin = std::min(gridsX[i], gridsX[i + 1]);
-          if (gridsY[i] == posY && xmin == posX && dir == 1) {
-            congestion_nets_.insert(nets_[netID]->getDbNet());
+          if (abs(gridsY[i] - posY) <= ratio && abs(xmin - posX) <= ratio
+              && dir == 1) {
+            congestion_nets.insert(nets_[netID]->getDbNet());
           }
         }
       }
     }
   }
 }
 
-int FastRouteCore::getOverflow2Dmaze(int* maxOverflow,
-                                     int* tUsage,
-                                     bool fillNetsVector)
+// The function will add the new nets to the congestion_nets set
+void FastRouteCore::getCongestionNets(std::set<odb::dbNet*>& congestion_nets)
+{
+  std::vector<int> xs, ys, dirs;
+  int n = 0;
+  // Find horizontal ggrids with congestion
+  for (int i = 0; i < y_grid_; i++) {
+    for (int j = 0; j < x_grid_ - 1; j++) {
+      const int overflow = h_edges_[i][j].usage - h_edges_[i][j].cap;
+      if (overflow > 0) {
+        xs.push_back(j);
+        ys.push_back(i);
+        dirs.push_back(1);
+        n++;
+      }
+    }
+  }
+  // Find vertical ggrids with congestion
+  for (int i = 0; i < y_grid_ - 1; i++) {
+    for (int j = 0; j < x_grid_; j++) {
+      const int overflow = v_edges_[i][j].usage - v_edges_[i][j].cap;
+      if (overflow > 0) {
+        xs.push_back(j);
+        ys.push_back(i);
+        dirs.push_back(0);
+        n++;
+      }
+    }
+  }
+
+  int old_size = congestion_nets.size();
+
+  for (int ratio = 0; ratio < 5 && old_size == congestion_nets.size();
```

Comment:
Could you explain what is the `ratio` variable, here and in `setCongestionNets`?

Reply:
The variable name is wrong, this variable serves to increase the radius around the congested zone in case new nets are not obtained.

Reply:
LGTM, but I'll let @maliberty give his review.